diff --git a/node_modules/react-native/react.gradle b/node_modules/react-native/react.gradle
index ff46476..c287ee6 100644
--- a/node_modules/react-native/react.gradle
+++ b/node_modules/react-native/react.gradle
@@ -170,6 +170,8 @@ afterEvaluate {
                 extraArgs.addAll(config.extraPackagerArgs)
             }
 
+            environment("APP_VARIANT", variant.name);
+
             commandLine(*execCommand, bundleCommand, "--platform", "android", "--dev", "${devEnabled}",
                 "--reset-cache", "--entry-file", entryFile, "--bundle-output", jsBundleFile, "--assets-dest", resourcesDir,
                 "--sourcemap-output", enableHermes ? jsPackagerSourceMapFile : jsOutputSourceMapFile, *extraArgs)
