diff --git a/node_modules/react-native/react.gradle b/node_modules/react-native/react.gradle
index 250299b..da1ceb2 100644
--- a/node_modules/react-native/react.gradle
+++ b/node_modules/react-native/react.gradle
@@ -291,6 +291,7 @@ afterEvaluate {
             if (config.extraPackagerArgs) {
                 extraArgs.addAll(config.extraPackagerArgs)
             }
+            environment("APP_VARIANT", variant.name);
 
             commandLine(*execCommand, bundleCommand, "--platform", "android", "--dev", "${devEnabled}",
                 "--reset-cache", "--entry-file", entryFile, "--bundle-output", jsBundleFile, "--assets-dest", resourcesDir,
