diff --git a/node_modules/avvio/plugin.js b/node_modules/avvio/plugin.js
index afa4375..826c829 100644
--- a/node_modules/avvio/plugin.js
+++ b/node_modules/avvio/plugin.js
@@ -11,7 +11,7 @@ function getName (func) {
   // let's see if this is a file, and in that case use that
   // this is common for plugins
   const cache = require.cache
-  const keys = Object.keys(cache)
+  const keys = [] // Object.keys(cache)
 
   // eslint-disable-next-line no-var
   for (var i = 0; i < keys.length; i++) {
