diff --git a/node_modules/bindings-noderify-nodejs-mobile/bindings.js b/node_modules/bindings-noderify-nodejs-mobile/bindings.js
index 472730b..4c1e584 100644
--- a/node_modules/bindings-noderify-nodejs-mobile/bindings.js
+++ b/node_modules/bindings-noderify-nodejs-mobile/bindings.js
@@ -38,7 +38,7 @@ var fs = require('fs'),
       ['module_root', 'node_modules', 'name', 'build', 'bindings'],
       // node-waf and gyp_addon (a.k.a node-gyp)
       ['module_root', 'node_modules', 'name', 'build', 'Debug', 'bindings'],
-      ['module_root', 'node_modules', 'name', 'build', 'Release', 'bindings'],
+      ['module_root', 'node_modules', 'better-sqlite3', 'build', 'Release', 'bindings'],
       // Debug files, for development (legacy behavior, remove for node v0.9)
       ['module_root', 'node_modules', 'name', 'out', 'Debug', 'bindings'],
       ['module_root', 'node_modules', 'name', 'Debug', 'bindings'],
